warehouse/accounts/views.py

```diff
@@ -96,11 +102,15 @@ def login(request, _form_class=LoginForm):
         # and we don't want to continue using the old one.
         request.session.new_csrf_token()
 
+        # If the user-originating redirection url is not safe, then redirect to
+        # the user's profile instead.
+        if not is_safe_url(url=redirect_to, host=request.host):
+            redirect_to = request.route_path("accounts.profile",
```

Comment:
I think it makes more sense to redirect users to the index view rather than their own profile in the case that the place they were trying to go was invalid. A route for this doesn't exist until #492 so for now you can just hardcode ``"/"``.